inc/global_assets/namespace.php

```diff
@@ -0,0 +1,343 @@
+<?php
+/**
+ * Altis Global Media.
+ *
+ * @package altis-media
+ */
+
+namespace Altis\Media\Global_Assets;
+
+use Altis;
+use Exception;
+use WP_Admin_Bar;
+use WP_Site;
+
+/**
+ * Setup global media hooks.
+ *
+ * @return void
+ */
+function bootstrap() {
+	$config = Altis\get_config()['modules']['media'];
+
+	add_action( 'plugins_loaded', __NAMESPACE__ . '\\load_amf', 9 );
+
+	// Load WP AMF and configure global site.
+	if ( empty( $config['global-media-library'] ) ) {
+		return;
+	}
+
+	// Bail if a site URL has been provided, assume it's external.
+	if ( is_string( $config['global-media-library'] ) ) {
+		define( 'AMF_WORDPRESS_URL', $config['global-media-library'] );
```

Comment:
If we're assuming it's on the same network it's gonna be somewhat flakier to allow folks to specify a site ID, it also makes the automation component harder to write. Better to handle that internally. I'm thinking about how portable setting a site ID would be between people's local set ups and cloud, even though it could be overwritten and would technically work ok even after importing a db.

Given we're moving the site creation stuff out to become a generic global content repository I think the best solution is still to require a fully qualified URL for an alternative media source. I don't feel there's much to gain from allowing multiple methods for specifying this and adds to the complexity and maintenance burden.

Reply:
Ah just realised too, in case it isn't clear the config option can just be set to `true` and it all gets handled in the background.